protocol/binary_test.go

```diff
@@ -788,3 +821,219 @@ func TestBinaryEnvelopeSuccessful(t *testing.T) {
 		assert.Equal(t, tt.encoded, buf.Bytes(), "%v: reencoded bytes mismatch")
 	}
 }
+
+func TestReqRes(t *testing.T) {
```

Comment:
Suggest renaming to describe the behavior being verified... perhaps `TestEnvelopeAgnosticism`...